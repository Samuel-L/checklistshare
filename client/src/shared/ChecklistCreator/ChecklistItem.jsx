import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from 'material-ui/styles';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import DeleteIcon from 'material-ui-icons/Delete';

const styles = () => ({
  root: {
    margin: '5px',
  },
  textField: {
    width: '90%',
  },
  icon: {
    minWidth: '5%',
    maxWidth: '10%',
    color: 'white',
    backgroundColor: 'lightCoral',
    marginLeft: '3px',
    padding: '2px 4px',
    '&:hover': {
      backgroundColor: 'red',
      boxShadow: 'none',
    },
  },
});

const ChecklistItem = props => (
  <div key={props.item.id} className={props.classes.root}>
    <TextField
      className={props.classes.textField}
      required
      id={`item-${props.item.id}`}
      label="Item"
      value={props.item.name}
      onChange={props.handleItemTextChange(props.idx)}
      margin="normal"
    />
    <Button
      className={props.classes.icon}
      id="delete-item-button"
      aria-label="delete"
      onClick={() => { props.handleItemDelete(props.idx); }}
      variant="raised"
    >
      <DeleteIcon />
    </Button>
  </div>
);

ChecklistItem.propTypes = {
  idx: PropTypes.number.isRequired,
  item: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
  }).isRequired,
  handleItemTextChange: PropTypes.func.isRequired,
  handleItemDelete: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(ChecklistItem);
