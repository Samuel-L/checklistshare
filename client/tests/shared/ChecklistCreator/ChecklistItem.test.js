import React from 'react';
import renderer from 'react-test-renderer';

import ChecklistItem from '../../../src/shared/ChecklistCreator/ChecklistItem';

describe('pages/Home/ChecklistCreator: ChecklistItem', () => {
  const item = { id: 0, name: 'Item' };
  const classes = { root: '', textField: '', icon: '' };
  const tree = renderer.create(
    <ChecklistItem
      idx={0}
      item={item}
      handleItemTextChange={jest.fn()}
      handleItemDelete={jest.fn()}
      classes={classes}
    />
).toJSON();

  it('should render', () => {
    expect(tree).toMatchSnapshot();
  });
});
