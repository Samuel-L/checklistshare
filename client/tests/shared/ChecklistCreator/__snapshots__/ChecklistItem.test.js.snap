// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Home/ChecklistCreator: ChecklistItem should render 1`] = `
<div
  className="ChecklistItem-root-1"
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-4 MuiFormControl-marginNormal-5 ChecklistItem-textField-2"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-13 MuiInputLabel-root-8 MuiInputLabel-formControl-9 MuiInputLabel-animated-12 MuiInputLabel-shrink-11"
      data-shrink={true}
      htmlFor="item-0"
    >
      Item
      <span
        className="MuiFormLabel-asterisk-17"
      >
        â€‰*
      </span>
    </label>
    <div
      className="MuiInput-root-18 MuiInput-formControl-19 MuiInput-underline-22"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-26"
        defaultValue={undefined}
        disabled={false}
        id="item-0"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="text"
        value="Item"
      />
    </div>
  </div>
  <button
    aria-label="delete"
    className="MuiButtonBase-root-46 MuiButton-root-31 MuiButton-raised-36 ChecklistItem-icon-3"
    disabled={false}
    id="delete-item-button"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-32"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-49"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-55"
    />
  </button>
</div>
`;
